client/errors.go

```diff
@@ -0,0 +1,142 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+// ErrorKind distinguishes kind of errors.
+type ErrorKind string
+
+// error kind const value doc comments here have a non-default,
+// specialized style (to help docs/error-kind.go):
+//
+// // ErrorKind...: DESCRIPTION [no final dot]
+//
+// `code-like` quoting should be used when meaningful.
+
+// Error kinds. Keep in sync with: https://forum.snapcraft.io/t/using-the-rest-api/18603#heading--errors
```

Comment:
Maybe we should mention the script your wrote to keep the errors in sync here? So that people don't keep this in sync manually?

Reply:
I did update this in the PR with this part of the change.